Library/Homebrew/test/test_diagnostic.rb

```diff
@@ -204,20 +204,6 @@ def test_check_for_symlinked_cellar
     HOMEBREW_CELLAR.mkpath
   end
 
-  def test_check_for_autoconf
-    MacOS::Xcode.stubs(:installed?).returns true
```

Comment:
An alternative would be to retain the test, but prefix it with `needs_compat`. See [`test_audit_class_github_gist_formula` from `test_cmd_audit.rb`](https://github.com/Homebrew/brew/blob/b53f0c5ada753d5ded6f6fbe4f920a4a43696394/Library/Homebrew/test/test_cmd_audit.rb#L253-L265) for an example.